File: backend/src/main/java/com/gpuoj/dto/ResultMessage.java
package com.gpuoj.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ResultMessage {
    private String submissionId;
    private String verdict;
    private String stdout;
    private String stderr;
    private Integer wallTimeMs;
    private Integer peakVramMb;
    private Double speedup;
    private String compileError;
}

File: backend/src/main/kotlin/com/gpuoj/dto/ResultMessage.kt
package com.gpuoj.dto

data class ResultMessage(
    val submissionId: String = "",
    val verdict: String? = null,
    val stdout: String? = null,
    val stderr: String? = null,
    val wallTimeMs: Int? = null,
    val peakVramMb: Int? = null,
    val speedup: Double? = null,
    val compileError: String? = null
)
